ls1intum/Artemis
# `Assessment`: Improve warnings for instructors before potentially assessing unfinished submissions

### Checklist
#### General


#### Server
#### Client

### Motivation and Context
Follow-up to #4416.

### Description
- Adds a warning on the ‘Submissions’ page if the regular due date is over but not the individual due date for some students.
- Adds a warning to the ‘Team’ page if the regular due date is over but not the individual due date for some students.
- Splits a method in `ProgrammingExerciseScheduleService` into two parts to satisfy Checkstyle (the split makes structural sense, too).
- Adds documentation how the individual due dates work when tests for the programming exercise are marked with visibility ‘After Due Date’.

### Steps for Testing

Prerequisites:
- An Instructor
- A team exercise of any type 

1. Submit something to the exercise as the student/team.
2. As instructor, set an individual due date to some point in the future for the team.
3. Set the regular due date to be in the past.
4. Go to the ‘Submissions‘ page and check that the warning as shown in the screenshot below is shown.
5. Go to the ‘Team’ detail page and check that the warning as shown in the second screenshot below is shown.

### Review Progress
#### Code Review
#### Manual Tests

### Test Coverage





| Component | Branch | Line |
| --- | --- | --- |
| `assessment-warning.component.ts` | 100% | 100% |

### Screenshots
Submissions page:
![Screenshot 2021-12-20 at 19-54-28 Assessment Dashboard](https://user-images.githubusercontent.com/64250573/146824828-23d60c9f-b6d6-4707-beb0-f5548aaaf126.png)

Team page:
![Screenshot 2021-12-20 at 19-54-08 Team](https://user-images.githubusercontent.com/64250573/146824839-42291ba2-67c4-4060-b7bc-6e3ce1af5513.png)
